tests/chainer_tests/functions_tests/evaluation_tests/test_classification_summary.py

```diff
@@ -0,0 +1,98 @@
+import unittest
+
+import numpy
+
+import chainer
+from chainer import cuda
+from chainer import functions as F
+from chainer import testing
+from chainer.testing import attr
+from chainer.testing import condition
+
+
+def recall(preds, ts, dtype, label_num):
+    tp = numpy.zeros((label_num,), dtype=numpy.int32)
+    support = numpy.zeros((label_num,), dtype=numpy.int32)
+    for p, t in zip(preds.ravel(), ts.ravel()):
+        support[t] += 1
+        if p == t:
+            tp[t] += 1
+    return dtype(tp) / support
+
+
+def precision(preds, ts, dtype, label_num):
+    tp = numpy.zeros((label_num,), dtype=numpy.int32)
+    relevant = numpy.zeros((label_num,), dtype=numpy.int32)
+    for p, t in zip(preds.ravel(), ts.ravel()):
+        relevant[p] += 1
+        if p == t:
+            tp[p] += 1
+    return dtype(tp) / relevant
+
+
+def f1_score(precision, recall, beta=1.0):
+    beta_square = beta * beta
+    return ((1 + beta_square) * precision * recall /
+            (beta_square * precision + recall))
+
+
+def support(ts, dtype, label_num):
+    ret = numpy.zeros((label_num,), dtype=numpy.int32)
+    for t in ts.ravel():
+        ret[t] += 1
+    return ret
+
+
+@testing.parameterize(
+    *testing.product_dict(
+        [{'y_shape': (30, 3), 't_shape': (30,)},
+         {'y_shape': (30, 3, 5), 't_shape': (30, 5)}],
```

Comment:
I found that the above settings sometimes raise RuntimeWarning because a denominator of F1 score may contains zero due to the small batchsize.
How about setting the batch size to a larger one (e.g., 100) to disable such annoying messages?